src/status_im/mailserver/core.cljs

```diff
@@ -485,22 +510,101 @@
                         {:topic topic
                          :mailserver-topic mailserver-topic})]})))
 
-(defn get-updated-mailserver-topics [db topics last-request]
-  (reduce (fn [acc topic]
-            (if-let [mailserver-topic (some-> (get-in db [:mailserver/topics topic])
-                                              (assoc :last-request last-request))]
-              (assoc acc topic mailserver-topic)
-              acc))
-          {}
-          topics))
+(defn recalculate-gap
```

Comment:
Probably it would be good to add some tests for this function, although +1 for the drawings :)

Reply:
sure, on it

Reply:
done